packages/verified-fetch/src/utils/responses.ts

```diff
@@ -96,3 +112,63 @@ export function movedPermanentlyResponse (url: string, location: string, init?:
 
   return response
 }
+
+type RangeOptions = { offset: number, length: number, totalSize?: number } | { offset: number, length?: number, totalSize: number }
+
+/**
+ * Some caveats about range responses here:
+ * * We only support single range requests (multi-range is optional), see https://specs.ipfs.tech/http-gateways/path-gateway/#range-request-header
+ * * Range responses are only supported for unixfs and raw data, see https://specs.ipfs.tech/http-gateways/path-gateway/#range-request-header.
+ *
+ * If the user requests something other than unixfs or raw data, we should not call this method and ignore the range header (200 OK). See https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests#partial_request_responses
+ *
+ * TODO: Supporting multiple range requests will require additional changes to the `handleDagPb` and `handleRaw` functions in `src/verified-fetch.js`
+ */
+export function okRangeResponse (url: string, body: SupportedBodyTypes, range: RangeOptions, init?: ResponseOptions): Response {
+  // if we know the full size of the body, we should use it in the content-range header. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range
+  let contentRangeHeader: string | undefined
+
+  try {
+    contentRangeHeader = getContentRangeHeader({ body, total: range.totalSize, offset: range.offset, length: range.length })
+  } catch (e) {
+    return badRangeResponse(url, body, init)
```

Comment:
It's probably worth logging the error to help ourselves in the future?

Reply:
done in [`05a6dfb` (#10)](https://github.com/ipfs/helia-verified-fetch/pull/10/commits/05a6dfb90631b527e078a4f9c61954c0a66e7905)